internal/test_cases/invalid_command_type_test_case.go

```diff
@@ -0,0 +1,37 @@
+package test_cases
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+type InvalidCommandTypeTestCase struct {
```

Comment:
I know I suggested this name, but looking at it now it is a bit confusing as to how `InvalidCommandTestCase` and `InvalidCommandTypeTestCase` are different. 

I wonder if we should call this a `CommandTypeTestCase` instead?  

```
func (t CommandTypeTestCase) RunForInvalidCommand()
func (t CommandTypeTestCase) RunForExecutable()
func (t CommandTypeTestCase) RunForBuiltin()
```

@andy1li Open to better names - ideally one should be able to look at the list of test cases within the `test_cases` package and roughly tell what they do even without looking at their implementations.